public/app/features/alerting/components/NewNotificationChannelForm.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FC, useEffect } from 'react';
+import { css } from 'emotion';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import {
+  Button,
+  Field,
+  FormAPI,
+  HorizontalGroup,
+  InfoBox,
+  Input,
+  InputControl,
+  Select,
+  stylesFactory,
+  Switch,
+  useTheme,
+} from '@grafana/ui';
+import { OptionElement } from './OptionElement';
+import { NotificationChannel, NotificationChannelDTO, Option } from '../../../types';
+
+interface Props extends Omit<FormAPI<NotificationChannelDTO>, 'formState'> {
+  selectableChannels: Array<SelectableValue<string>>;
+  selectedChannel?: NotificationChannel;
+  imageRendererAvailable: boolean;
+
+  onTestChannel: (data: NotificationChannelDTO) => void;
+}
+
+export const NewNotificationChannelForm: FC<Props> = ({
+  control,
+  errors,
+  selectedChannel,
+  selectableChannels,
+  register,
+  watch,
+  getValues,
+  imageRendererAvailable,
+  onTestChannel,
+}) => {
+  const styles = getStyles(useTheme());
+
+  useEffect(() => {
+    watch(['type', 'settings.priority', 'sendReminder', 'uploadImage']);
+  }, []);
+
+  return (
+    <>
+      <div className={styles.basicSettings}>
+        <Field label="Name" invalid={!!errors.name} error={errors.name && errors.name.message}>
+          <Input name="name" ref={register({ required: 'Name is required' })} />
+        </Field>
+        <Field label="Type">
+          <InputControl
+            name="type"
+            as={Select}
+            options={selectableChannels}
+            control={control}
+            rules={{ required: true }}
+          />
+        </Field>
+        <Field label="Default" description="Use this notification for all alerts">
+          <Switch name="isDefault" ref={register} />
+        </Field>
+        <Field label="Include image" description="Captures an image and include it in the notification">
+          <Switch name="settings.uploadImage" ref={register} />
+        </Field>
+        {getValues().uploadImage && !imageRendererAvailable && (
```

Comment:
`getValues()` gets evaluated multiple times, so it'd make sense to save it to a const (before the render()) and reuse to respect DRY, i.e.:
`const formValues = getValues();`

Reply:
yes, was about to write the same.